src/sciline/pipeline.py

```diff
@@ -322,6 +322,9 @@ def __init__(
         params:
             Dictionary of concrete values to provide for types.
         """
+        self._copy_providers = list(providers or [])
+        self._copy_params: Dict[type, Any] = {}
```

Comment:
I don't understand why those are needed? Can't you just make copies of `self._providers`, etc.?

Reply:
No because `self._providers` contains all the original providers, plus a whole bunch of lambda functions that were made from the parameters, which then fail if you try to construct a new pipeline with them.

Also, going through the sub-providers is a bit of a mess, especially with generics. So I thought just saving the constructor params was far easier. But maybe @SimonHeybrock has a better idea?

Reply:
Why are the lambdas failing?

Reply:
`ValueError: Provider <function Pipeline.__setitem__.<locals>.<lambda> at 0x7f492825f130> lacks type-hint for return value`

Reply:
The lambda is raising that? That is really odd.

Reply:
Just to be clear, I get the error from the lambdas when I do:
```Py
Pipeline(existing_pipeline._providers.values())
```

Reply:
Making a copy of the dicts internally works fine.